tests/unit_tests/test_import_gaurds.py

```diff
@@ -0,0 +1,98 @@
+# Standard Python modules
+import sys
+import unittest
+from unittest.mock import patch
+import warnings
+
+
+def ignore_warnings(test_func):
```

Comment:
Nice

Reply:
Yeah you can see my comment above, but this is because numpy is annoying for these tests.